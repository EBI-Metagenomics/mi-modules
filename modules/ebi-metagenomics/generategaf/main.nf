
process GENERATEGAF {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:0.2.0--pyhdfd78af_0':
        'biocontainers/mgnify-pipelines-toolkit:0.2.0--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(ips)

    output:
    tuple val(meta), path("*.gaf"), emit: gaf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    generate_gaf \\
        -i ${ips} \\
        -o ${prefix} \\

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_ips_annotations.gaf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """
}
