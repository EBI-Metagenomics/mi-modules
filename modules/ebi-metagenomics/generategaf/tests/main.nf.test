
nextflow_process {

    name "Test Process GENERATEGAF"
    script "../main.nf"
    process "GENERATEGAF"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "generategaf"

    test("generategaf") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/ips_out.tsv", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(process.out.gaf.get(0).get(1)).text.contains("EMG	GO:0042773	GO		GO:0042773")).match("gaf_test_1") },
                { assert snapshot(path(process.out.gaf.get(0).get(1)).text.contains("EMG	GO:0003723	GO		GO:0003723")).match("gaf_test_2") },
                { assert snapshot(path(process.out.gaf.get(0).get(1)).text.contains("EMG	GO:0003700	GO		GO:0003700")).match("gaf_test_3") }
            )
        }

    }

    test("generategaf - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/ips_out.tsv", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
