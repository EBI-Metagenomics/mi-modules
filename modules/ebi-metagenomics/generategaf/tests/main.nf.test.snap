{
    "gaf_test_3": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-27T10:37:05.471621383"
    },
    "gaf_test_2": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-27T10:37:05.448280559"
    },
    "gaf_test_1": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-27T10:37:05.426270774"
    },
    "generategaf - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_ips_annotations.gaf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,53ff6fdbb93140587a50a5ed8aa4aa8f"
                ],
                "gaf": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_ips_annotations.gaf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,53ff6fdbb93140587a50a5ed8aa4aa8f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-27T10:37:11.463914247"
    }
}